### PR TITLE
SUBS 688 and 689 archive refactor

### PR DESCRIPTION
**Pull request by davidrichardson**
SUBS-688 and SUBS-689
- add component to assign archives
- refactor dispatcher to use assigned archive

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -79,7 +79,7 @@ uploadArchives {
                 url 'https://github.com/EMBL-EBI-SUBS/subs-dispatcher'
 
                 scm {
-                    url 'https://github.com/EMBL-EBI-SUBS/subs-dipatcher'
+                    url 'https://github.com/EMBL-EBI-SUBS/subs-dispatcher'
```
Comment (fpenim) [inline build.gradle]: It's missing the `.git`, should be:
`url 'https://github.com/EMBL-EBI-SUBS/subs-dispatcher.git'`

File: src/main/java/uk/ac/ebi/subs/processing/archiveassignment/SubmissionArchiveAssignmentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.subs.data.component.Archive;
+import uk.ac.ebi.subs.processing.dispatcher.SubmissionEnvelopeService;
+import uk.ac.ebi.subs.processing.archiveassignment.assigners.StoredSubmittableArchiveAssignmentService;
+import uk.ac.ebi.subs.repository.model.ProcessingStatus;
+import uk.ac.ebi.subs.repository.model.Submission;
+import uk.ac.ebi.subs.repository.repos.status.ProcessingStatusRepository;
+
+@Component
+public class SubmissionArchiveAssignmentServiceImpl implements SubmissionArchiveAssignmentService {
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/processing/archiveassignment/SubmissionArchiveAssignmentServiceImpl.java]: If this is the only class implementing the SubmissionArchiveAssignmentService Interface, why do we need the Interface?

File: src/main/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import uk.ac.ebi.subs.data.Submission;
+import uk.ac.ebi.subs.processing.SubmissionEnvelope;
+import uk.ac.ebi.subs.repository.model.StoredSubmittable;
+import uk.ac.ebi.subs.repository.repos.SubmissionRepository;
+import uk.ac.ebi.subs.repository.repos.submittables.*;
+
+import java.util.stream.Stream;
+
+@Service
+public class SubmissionEnvelopeServiceImpl implements SubmissionEnvelopeService {
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceImpl.java]: Again, do we need an Interface that is only implemented once?

File: src/test/java/uk/ac/ebi/subs/processing/archiveassigner/ArchiveAssignmentTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import uk.ac.ebi.subs.repository.repos.submittables.SampleRepository;
+import uk.ac.ebi.subs.repository.repos.submittables.StudyRepository;
+import uk.ac.ebi.subs.repository.services.SubmissionHelperService;
+import uk.ac.ebi.subs.repository.services.SubmittableHelperService;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringRunner.class)
```
Comment (fpenim) [inline src/test/java/uk/ac/ebi/subs/processing/archiveassigner/ArchiveAssignmentTest.java]: Why different tests use different 'runners' ?
SubmissionEnvelopeServiceTest uses:`@RunWith(SpringJUnit4ClassRunner.class)`

File: src/test/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import uk.ac.ebi.subs.repository.services.SubmittableHelperService;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
```
Comment (fpenim) [inline src/test/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceTest.java]: Different test runner `@RunWith(SpringRunner.class)` from ArchiveAssignmentTest ?

File: src/test/java/uk/ac/ebi/subs/processing/archiveassigner/ArchiveAssignmentTest.java (hunk 4 above)
Comment (davidrichardson) [reply src/test/java/uk/ac/ebi/subs/processing/archiveassigner/ArchiveAssignmentTest.java]: I'll make them consistent. SpringRunner is an alias for SpringJUnit4ClassRunner.

File: src/main/java/uk/ac/ebi/subs/processing/archiveassignment/SubmissionArchiveAssignmentServiceImpl.java (hunk 2 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/processing/archiveassignment/SubmissionArchiveAssignmentServiceImpl.java]: true, I'll refactor it

File: src/main/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceImpl.java (hunk 3 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/processing/dispatcher/SubmissionEnvelopeServiceImpl.java]: agreed

**Pull request merged**